### PR TITLE
feat(react): add usePowerSyncStatus hook

### PR DESCRIPTION
**Pull request by DominicGBauer**
## Description
Add usePowerSyncStatus hook to react package

## Work Done
* Added `usePowerSyncStatus`
* Added testing to react package
* Updated demo to use new hook
* Removed unused imports from web SDK tests

## How to test
* Load `react-supabase-todolist` and make sure Wifi icon is changing is expected (i.e. online when connected and offline when disconnected)

### CODE REVIEW

File: demos/react-supabase-todolist/src/app/views/layout.tsx [hunk 1]
```diff
@@ -25,16 +25,16 @@ import React from 'react';
 
 import { useNavigationPanel } from '@/components/navigation/NavigationPanelContext';
 import { useSupabase } from '@/components/providers/SystemProvider';
-import { usePowerSync } from '@journeyapps/powersync-react';
+import { usePowerSync, usePowerSyncStatus } from '@journeyapps/powersync-react';
```
Comment (Chriztiaan) [inline demos/react-supabase-todolist/src/app/views/layout.tsx:28]: Nice. There should be a similar code example in the react-native project.

Comment (Chriztiaan) [reply demos/react-supabase-todolist/src/app/views/layout.tsx:28]: See `HeaderWidget.tsx` under rn-supabase-todolist.

**Comment by Chriztiaan** [comment]
Last thing I missed was adding an entry to the React package's readme. We have examples of all the other hooks there.
Tested, verified working.

**Review by Chriztiaan — DISMISSED** [review]

File: packages/powersync-react/tests/PowerSyncContext.test.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+describe('PowerSyncContext', () => {
+  describe('usePowerSync', () => {
+    it('should retrieve the PowerSync DB from the context and display the test text', async () => {
+      const TestComponent = () => {
+        const powerSyncDb = SUT.usePowerSync() as any;
+        return <div>{powerSyncDb.testText}</div>;
+      };
+
+      const { findByText } = render(
+        <SUT.PowerSyncContext.Provider value={{ testText: 'Test Text' } as any}>
```
Comment (stevensJourney) [inline packages/powersync-react/tests/PowerSyncContext.test.tsx:15]: This kinda feels like it's testing that the React context provider is working as a React context provider. I don't think we need to test for that.

File: packages/powersync-react/tsconfig.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "baseUrl": "./",
     "esModuleInterop": true,
     "jsx": "react",
-    "rootDir": "src",
+    "rootDir": "./",
```
Comment (rkistner) [inline packages/powersync-react/tsconfig.json:6]: Why this change? Most of our packages use `"rootDir": "src"`, and it gives a nicer output structure

Comment (DominicGBauer) [reply packages/powersync-react/tsconfig.json:6]: I created a test folder which is outside the root and was getting build errors

File: packages/powersync-react/tests/PowerSyncContext.test.tsx (hunk 2 above)
Comment (DominicGBauer) [reply packages/powersync-react/tests/PowerSyncContext.test.tsx:15]: I think it's useful if to catch an inadvertent change PowerSyncContext and since the test costs next to nothing it worth having IMO.

File: packages/powersync-react/tsconfig.json (hunk 3 above)
Comment (DominicGBauer) [reply packages/powersync-react/tsconfig.json]: Nvm I'm a noob, reverted this back

File: packages/powersync-react/tests/PowerSyncContext.test.tsx (hunk 2 above)
Comment (stevensJourney) [reply packages/powersync-react/tests/PowerSyncContext.test.tsx:15]: That's fair enough 👍

**Review by rkistner — APPROVED** [review]

**Review by Chriztiaan — APPROVED** [review]

**Review by stevensJourney — APPROVED** [review]
